src/mlpack/core/data/confusion_matrix.hpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file confusion_matrix.hpp
+ * @author Jeffin Sam
+ *
+ * Function for confusion matrix
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_CONFUSION_MATRIX_HPP
+#define MLPACK_CORE_DATA_CONFUSION_MATRIX_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+/**
+ * A confusion matrix is a summary of prediction results on a classification
+ * problem.The number of correct and incorrect predictions are summarized
+ * with count values and broken down by each class.
+ *
+ * @param pred vector of predicted values.
+ * @param actual vector of actual values.
+ * @param output matrix which is represented as confusion matrix.
+ * @param countlables no of classes
+ * for example for 2 classes the function will be
+ * confusionmatrix(pred,actual,matrix,2)
```

Comment:
We might want to update the parameter names here as well, also there is a missing space after the comma.

Reply:
yes updated